Cilsil/Utils/ProgramState.cs

```diff
@@ -191,13 +191,15 @@ public class ProgramState
         /// </summary>
         /// <param name="method">The method being translated.</param>
         /// <param name="cfg">The CFG associated with the method.</param>
-        public ProgramState(MethodDefinition method, Cfg cfg)
+        /// <param name="guardian">If <c>true</c>, record resource leak location by locating exit node  
+        /// at the location of the very first instruction of the corresponding method.</param>
+        public ProgramState(MethodDefinition method, Cfg cfg, bool guardian)
         {
```

Comment:
Make this an optional parameter. Given that this is a rare use case you should not be adding a mandatory param.

Reply:
With default value false